Zotero/Scenes/Detail/PDF/Views/Annotation View/AnnotationView.swift

```diff
@@ -97,14 +97,33 @@ final class AnnotationView: UIView {
     /// - parameter selected: If true, selected state style is applied.
     /// - parameter availableWidth: Available width for view.
     /// - parameter library: Library of given annotation
+    /// - parameter pdfAnnotationsCoordinatorDelegate: Delegate for getting share menu.
+    /// - parameter state: State required for setting up share menu.
     func setup(with annotation: Annotation, comment: Comment?, preview: UIImage?, selected: Bool, availableWidth: CGFloat, library: Library, currentUserId: Int, displayName: String, username: String,
-               boundingBoxConverter: AnnotationBoundingBoxConverter) {
+               boundingBoxConverter: AnnotationBoundingBoxConverter, pdfAnnotationsCoordinatorDelegate: PdfAnnotationsCoordinatorDelegate, state: PDFReaderState) {
         let editability = annotation.editability(currentUserId: currentUserId, library: library)
         let color = UIColor(hex: annotation.color)
         let canEdit = editability == .editable && selected
 
-        self.header.setup(with: annotation, libraryId: library.identifier, isEditable: (editability != .notEditable && selected), showsLock: editability != .editable,
-                          showDoneButton: self.layout.showDoneButton, accessibilityType: .cell, displayName: displayName, username: username)
+        var annotationIsShareable = false
+        if let button = header.shareButton,
+           let menu = pdfAnnotationsCoordinatorDelegate.shareAnnotationMenu(state: state, annotation: annotation, sender: button)
```

Comment:
This is third time I see this when setting up the header. Can this be included in the `self.header.setup(...)` itself? Maybe pass flag whether it is shareable and also coordinator delegate so that it can create the menu by itself?

Reply:
Same as above, check https://github.com/zotero/zotero-ios/pull/693/commits/fe799ee2da8c0add7f8d4d82e027864e3f0bf79d